### ISSUE TITLE
Architecture - Big No Brainer Network

### ISSUE THREAD

**Issue by jayfranck** [OrsoEric/kaggle-Lux-AI#12]
NN on SL without prior knowledge of any kind

**Comment by OrsoEric** [comment]
A single network takes in input observations in martix forms and emit one matrix for each possible action TYPE, heatmap of probability of action. A translation block emits action where units and city tiles are on map.
![image](https://user-images.githubusercontent.com/30684972/135760728-54837049-6328-4efe-9356-0d1399dfb198.png)

**Comment by dgcrox** [comment]
Task neede to implement the BOT:  

**Network Architecture:**
- [x]  Define the input matrices https://github.com/OrsoEric/kaggle-Lux-AI/issues/16
- [ ]  Define the output matrices https://github.com/OrsoEric/kaggle-Lux-AI/issues/18
- [ ]  Define the core network (RNN?) 

**Inference**
- [x] Write the function that from a Game() structure return the input matricies https://github.com/OrsoEric/kaggle-Lux-AI/issues/15
- [ ] Create a translator to convert output to game action https://github.com/OrsoEric/kaggle-Lux-AI/issues/17

**Training Supervised Learning**  

- [ ] #22 
- [ ] #23 

**Training Reinforcemet Learning**  

- [ ] Define how the network is trained and how to integrate the game simulator https://github.com/OrsoEric/kaggle-Lux-AI/issues/19
- [ ] Write the training reward function

**Timeline**  

- [x] 2021-10-10 Complete input component of the BOT
- [ ] 2021-10-17 Complete output component of the BOT and generate supervised training data from top Kaggle Bots
- [ ] 2021-10-24 Submit supervised learning version of the BOT @OrsoEric  @dgcrox  @jayfranck

**Comment by jayfranck** [comment]
![project](https://user-images.githubusercontent.com/91155110/136663124-91117e87-3dd6-43ba-899c-5a37f62c8fe4.png)

**Comment by jayfranck** [comment]
https://www.tensorflow.org/tutorials/images/cnn

**Comment by OrsoEric** [comment]
Input stage of BNB implemented and tested
